dynamiqs/hermitian_conjugate.py

```diff
@@ -0,0 +1,23 @@
+from jaxtyping import ArrayLike
+from qutip import Qobj
+
+from .qarrays import QArray, QArrayLike, asqarray
+
+
+class HermitianConjugate:
+    def __radd__(self, y: QArrayLike) -> QArrayLike:
+        if isinstance(y, Qobj):
+            y = asqarray(y)
+
+        if isinstance(y, QArray):
+            return y + y.dag()
+        elif isinstance(y, ArrayLike):
+            return y + y.mT.conj()
+        else:
```

Comment:
The QArrayLike type also includes nested sequences of QArrayLike. So I guess there's something a bit subtle to handle here. Maybe we could only support `dq.hc` with QArrays to begin with ?

Reply:
There is no `QArrayLike` here I think so we're good

Reply:
Not sure what you mean @abocquet? If the input type is `QArrayLike`, the object is not either a `QArray` or an `ArrayLike`, see https://github.com/dynamiqs/dynamiqs/blob/main/dynamiqs/qarrays/qarray.py#L578-L594 for the complete definition of `QArrayLike`.